valet/workfunc.go

```diff
@@ -496,19 +519,26 @@ func makeWork(path FilePath, plan WorkPlan) (Work, error) {
 }
 
 func createMD5File(path string, md5sum []byte) (err error) { // NRV
-	f, err := os.OpenFile(path, os.O_RDWR|os.O_CREATE|os.O_EXCL, 0666)
-	if err != nil {
-		err = errors.Wrap(err, "will not overwrite an existing MD5 file")
+	var f *os.File
+	if f, err = ioutil.TempFile(os.TempDir(), "valet-"); err != nil {
+		return
+	}
+
+	_, err = f.WriteString(fmt.Sprintf("%x\n", md5sum))
+	if err = f.Close(); err != nil {
 		return
 	}
 
 	defer func() {
-		err = utilities.CombineErrors(err, f.Close())
+		// Clean up if we got this far and the temp file still exists
+		if rerr := os.Remove(f.Name()); !os.IsNotExist(rerr) {
+			err = utilities.CombineErrors(err, rerr)
+		}
 	}()
 
-	encoded := make([]byte, hex.EncodedLen(len(md5sum)))
-	hex.Encode(encoded, md5sum)
-	_, err = f.Write(append(encoded, '\n'))
+	if err = os.Rename(f.Name(), path); err != nil {
```

Comment:
No need for the `if`. Just return err after `Rename`. And does Rename have the same behaviour as the previous implementation? Ie. does it fail if the file exists? Is it still desirable to not overwrite an existing md5 file?

Reply:
I see, thanks. That's better.

I changed my mind about the overwrite behaviour - I'd rather have a change in the local file always propagate to the checksum file, rather than have a job start failing until a human came and checked what had happened.